### DISCUSSION TITLE
LSP signature help placement (to avoid obstructing view)

### DISCUSSION THREAD

**Discussion by pacien** [helix-editor/helix#8972] [General]
I find the information provided by the LSP signature help very useful.

However, the popup obstructs the view of the current context,
which makes it difficult to pick symbols as arguments, defeating the point.

Is there a way to display the signature help elsewhere,
such as (in preferred order):

- at the top right corner, like for LSP lints?
- in the status bar?
- at the top just above or below the bufferline?
- as virtual text, shifting the content instead of covering it?

For now, I have resorted to disabling the automatic popup with

```ini
[editor.lsp]
auto-signature-help = false
```

… and showing it on-demand with a manual keyboard shortcut,
but this only reduces the frequency of the annoyance.

**Comment by ghost** [comment]
Would love if this was customizable the same way as editor.inline-diagnostics

**Comment by Moritz-JD-Krueger** [comment]
I agree. I mainly use Helix for LaTeX and Quarto editing, but I can barely read the LSP popups because they appear right where I have a lot of text.

**Comment by qezz** [comment]
With this patch you can move the popup to be below the cursor

```diff
diff --git a/helix-term/src/handlers/signature_help.rs b/helix-term/src/handlers/signature_help.rs
index 8a0c9754..7381ed72 100644
--- a/helix-term/src/handlers/signature_help.rs
+++ b/helix-term/src/handlers/signature_help.rs
@@ -268,7 +268,7 @@ pub fn show_signature_help(

     let mut popup = Popup::new(SignatureHelp::ID, contents)
         .position(old_popup.and_then(|p| p.get_position()))
-        .position_bias(Open::Above)
+        .position_bias(Open::Below)
         .ignore_escape_key(true);

     // Don't create a popup if it intersects the auto-complete menu.
```

<img width="364" height="155" alt="image" src="https://github.com/user-attachments/assets/d04abe9d-2f89-4cc0-8971-036672b5770e" />

Didn't have time to make it configurable yet, but seems to be relatively easy

**Reply by qezz** [reply]
I opened a PR with this patch and a config option
- https://github.com/helix-editor/helix/pull/14215

**Comment by johannesrave** [comment]
yes please!
